build/dockerfiles/dev.Dockerfile

```diff
@@ -6,81 +6,49 @@
 # SPDX-License-Identifier: EPL-2.0
 #
 
-# https://quay.io/eclipse/che-machine-exec#^7\.
-FROM quay.io/eclipse/che-machine-exec:7.56.0 as machine-exec
+FROM quay.io/devfile/universal-developer-image:latest
```

Comment:
@vitaliy-guliy 
As I mentioned in your previous PR https://github.com/che-incubator/che-code/pull/246#discussion_r1245414076
I don't think we should bring UDI for the `Che-Code` development

But in general it's OK for me to merge the current PR, if the changes look good for other reviewers

Reply:
The benefit of using UDI as the base image is that all UDI layers are already in the nodes of the cluster (especially if kube-image-puller is configured, so it should be really fast to pull images based on UDI on a kube cluster.